File: backend-v5/lms/projects/ProjectParams.php
<?php

// Modul: lms/projects
// Klasa: ProjectParams
// Opis: parametri projekata na predmetu

require_once(Config::$backend_path."core/DB.php");

class ProjectParams {
	public $id;
	public $courseUnitId, $academicYearId, $minTeams, $maxTeams, $minTeamMembers, $maxTeamMembers, $locked;
	
	// Gets just one poll that is active for all courses in current academic year 
	// (usually there aren't more such polls)
	public static function fromCourse($courseUnitId, $academicYearId) {
		$q10 = DB::query("select min_timova, max_timova, min_clanova_tima, max_clanova_tima, zakljucani_projekti from predmet_projektni_parametri where predmet=$courseUnitId and akademska_godina=$academicYearId");
		if (mysql_num_rows($q10) < 1) {
			throw new Exception("no params set for project");
		}

		$pp = new ProjectParams;
		$pp->courseUnitId = $courseUnitId;
		$pp->academicYearId = $academicYearId;
		$pp->minTeams = mysql_result($q10,0,0);
		$pp->maxTeams = mysql_result($q10,0,1);
		$pp->minTeamMembers = mysql_result($q10,0,2);
		$pp->maxTeamMembers = mysql_result($q10,0,3);
		if (mysql_result($q10,0,4) == 1) $pp->locked = true; else $pp->locked = false;
		
		return $pp;
	}
}

?>

File: rorbackend/app/models/lms/project/project_params.rb
class Lms::Project::ProjectParams < ActiveRecord::Base
  # Uncomment following lines if working with legacy database
  # set_table_name 'predmet_projektni_parametri'
  # set_primary_key :predmet, :akademska_godina
  # alias_attribute :course_unit_id, :predmet
  # alias_attribute :academic_year_id, :akademska_godina
  # alias_attribute :min_teams, :min_timova
  # alias_attribute :max_teams, :max_timova
  # alias_attribute :min_team_members, :min_clanova_tima
  # alias_attribute :max_team_members, :max_clanova_tima
  # alias_attribute :locked, :zakljucani_projekti

  # Uncomment following lines if working with legacy database
  # TABLE_NAME = 'predmet_projektni_parametri'
  # COURSE_UNIT_ID = TABLE_NAME + '.' + 'predmet'
  # ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'akademska_godina'
  # MIN_TEAMS = TABLE_NAME + '.' + 'min_timova'
  # MAX_TEAMS = TABLE_NAME + '.' + 'max_timova'
  # MIN_TEAM_MEMBERS = TABLE_NAME + '.' + 'min_clanova_tima'
  # MAX_TEAM_MEMBERS = TABLE_NAME + '.' + 'max_clanova_tima'
  # LOCKED = TABLE_NAME + '.' + 'zakljucani_projekti'

  # Comment following lines if working with legacy database
  TABLE_NAME = 'lms_project_project_params'
  COURSE_UNIT_ID = TABLE_NAME + '.' + 'course_unit_id'
  ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'academic_year_id'
  MIN_TEAMS = TABLE_NAME + '.' + 'min_teams'
  MAX_TEAMS = TABLE_NAME + '.' + 'max_teams'
  MIN_TEAM_MEMBERS = TABLE_NAME + '.' + 'min_team_members'
  MAX_TEAM_MEMBERS = TABLE_NAME + '.' + 'max_team_members'
  LOCKED = TABLE_NAME + '.' + 'locked'

  ALL_COLUMNS = [COURSE_UNIT_ID, ACADEMIC_YEAR_ID, MIN_TEAMS, MAX_TEAMS, MIN_TEAM_MEMBERS, MAX_TEAM_MEMBERS, LOCKED]
  
  belongs_to :course_unit, :class_name => "Core::CourseUnit"
  belongs_to :academic_year, :class_name => "Core::AcademicYear"
  
  
  def self.from_course(course_unit_id, academic_year_id)
    project_params = (Lms::Project::ProjectParams).where(:course_unit_id => course_unit_id, :academic_year_id => academic_year_id)
    
    return project_params
  end
  
  
end
